rasa/core/channels/twilio_voice.py

```diff
@@ -0,0 +1,116 @@
+import inspect
+from sanic import Blueprint, response
+from sanic.request import Request
+from sanic.response import HTTPResponse
+from twilio.twiml.voice_response import VoiceResponse, Gather
+from typing import Text, Callable, Awaitable, List
+
+from rasa.shared.core.events import BotUttered
+from rasa.core.channels.channel import (
+    InputChannel,
+    CollectingOutputChannel,
+    UserMessage,
+)
+
+
+class TwilioVoiceInput(InputChannel):
+    @classmethod
+    def name(cls) -> Text:
+        """Name of your custom channel."""
+        return "twilio_voice"
+
+    def blueprint(
+        self, on_new_message: Callable[[UserMessage], Awaitable[None]]
+    ) -> Blueprint:
+
+        twilio_voice_webhook = Blueprint(
+            "custom_webhook_{}".format(type(self).__name__),
+            inspect.getmodule(self).__name__,
+        )
+
+        @twilio_voice_webhook.route("/", methods=["GET"])
+        async def health(request: Request) -> HTTPResponse:
+            return response.json({"status": "ok"})
+
+        @twilio_voice_webhook.route("/webhook", methods=["POST"])
+        async def receive(request: Request):
+            sender_id = request.form.get("From")  # method to get sender_id
+            text = request.form.get("SpeechResult")  # method to fetch text
+            input_channel = self.name()  # method to fetch input channel
+            call_status = request.form.get("CallStatus")  # method to fetch call status
+            call_sid = request.form.get("CallSid")  # Get the call identifier
+
+            collector = TwilioVoiceCollectingOutputChannel()
+
+            # Provide an initial greeting to answer the user's call.
+            if (text is None) & (call_status == "ringing"):
+                text = "hello"
+
+            # determine the response.
+            if text is not None:
+                await on_new_message(
+                    UserMessage(
+                        text, collector, sender_id, input_channel=input_channel,
+                    )
+                )
+
+                # Parse the text responses and build the Twilio VoiceResponse.
+                respond_segments = []
+                for message in collector.messages:
+                    respond_segments.append(message["text"])
```

Comment:
How about overriding the `OutputChannel.send_***` methods for this? Seems preferrable to me over trying to detect the message type manually.

Reply:
I created `TwilioVoiceCollectingOutputChannel` to override the button and image methods. In the voice channel the images will be removed.